src/sentry/reprocessing.py

```diff
@@ -1,5 +1,34 @@
 from __future__ import absolute_import
 
+import uuid
+
+
+REPROCESSING_OPTION = 'sentry:processing-rev'
+
+
+def get_reprocessing_revision(project, cached=True):
+    """Returns the current revision of the projects reprocessing config set."""
+    from sentry.models import ProjectOption, Project
+    if cached:
+        return ProjectOption.objects.get_value(project, REPROCESSING_OPTION)
+    try:
+        if isinstance(project, Project):
```

Comment:
Perhaps instead of having these 2 branches you could do something like
```
if not cached:
    ProjectOption.reload_cache(project.id)
ProjectOption.objects.get_value(project, REPROCESSING_OPTION)
```

Reply:
@alex-hofsteede I did not want to invalidate the cache because that seems questionable. Right now code seems to want to see a consistent view of the options. If we start to reload the cache randomly there some assumptions might be invalidated. The only reason we bypass the cache for this value is that we need it here specifically to refer to the latest but only in that one line.

Reply:
But if the cache is not invalidated, won't the `reprocessing_rev` pick up the old value again the next time round?

Reply:
ie, would we get in an infinite reprocessing loop because the value we get for `reprocessing_rev` (which always uses the cache) is always different from the value of `get_reprocessing_revision(project_id, cached=False)` (which doesn't) ?

Reply:
@alex-hofsteede so in theory this should happen:

Upon starting of processing we fetch the current state of the processing rev. If concurrently someone pushes new debug symbols up that is going to be changed. However all code futher along the thread of execution will see the original value still. If we hit another processing error and we are going to reprocessing, we only create a processing issue and raw event if the revision rev is still the same (uncached access).

Not sure if that makes sense? Should never go into an endless loop because the cache is only valid for the current thread of execution until task end.